File: phase2/old/state-event/newmodel/fornow/v1/jsontomermaid.py
import json

# Load JSON data from a file
with open("v1-step10-correct-4th.json", "r", encoding="utf-8") as f:
    graph_data = json.load(f)

# Initialize Mermaid diagram
mermaid_code = "```mermaid\ngraph TD;\n"

# Add nodes
node_map = {}  # Store node descriptions
for node in graph_data["nodes"]:  # Directly access "nodes" from the top level
    node_id = node["id"]
    description = node["description"]
    node_map[node_id] = description
    mermaid_code += f'  {node_id}["{description}"];\n'

# Add edges
for edge in graph_data["edges"]:  # Directly access "edges" from the top level
    from_node = edge["from"]  # Use "from" from JSON (not "from_node")
    to_node = edge["to"]
    edge_label = edge["description"]
    mermaid_code += f'  {from_node} -->|"{edge_label}"| {to_node};\n'

# Close the Mermaid code block
mermaid_code += "```\n"

# Save to file
with open("mermaid.md", "w", encoding="utf-8") as f:
    f.write(mermaid_code)

print("Mermaid diagram saved to mermaid.md!")

File: phase2/old/0416/v4/jsontomermaid.js
const fs = require("fs");

function convertJsonToMermaid(inputFile, outputFile) {
  if (!fs.existsSync(inputFile)) {
    console.error(`Input file ${inputFile} not found!`);
    return;
  }

  const rawData = fs.readFileSync(inputFile, "utf-8");
  const graphData = JSON.parse(rawData);

  let mermaidCode = "graph TD\n";  // Initialize Mermaid code for flowchart

  // Process nodes (both states and events)
  graphData.nodes.forEach(node => {
    let nodeId = node.id;
    let description = node.properties.description || '';

    // Add nodes to the Mermaid diagram
    if (node.type === 'state' || node.type === 'event') {
      mermaidCode += `    ${nodeId}\n`;

    }
  });

// Process edges
graphData.edges.forEach(edge => {
  let fromState = edge.from;
  let toState = edge.to;

  let action = edge.action || '';
  let condition = edge.condition || '';

  // Use action as label, if action is missing use condition
  let label = action || condition;

  // Add the edge (transition) between nodes to the Mermaid diagram
  mermaidCode += `    ${fromState} -->|"${label}"| ${toState}\n`;
});

  // Wrap the Mermaid code in markdown syntax for Mermaid rendering
  mermaidCode = "```mermaid\n" + mermaidCode + "```";

  // Write the Mermaid code to the output file
  fs.writeFileSync(outputFile, mermaidCode, "utf-8");
  console.log(`Mermaid diagram saved to ${outputFile}`);
}

// Example usage
const inputFile = "step1-v2.json"; // Absolute path
const outputFile = "converted-mermaid-v2.md";  // Output file path
convertJsonToMermaid(inputFile, outputFile);

